File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/Solution.cpp
#include <string>
using namespace std;

class Solution {
public:
    int minimumRecolors(string blocks, int k)
    {
        int cnt_W = 0, i = 0, n = blocks.size();
        for (; i < k; i++)
            cnt_W += blocks[i] == 'W';

        int res = cnt_W;
        for (; i < n; i++) {
            cnt_W -= blocks[i - k] == 'W';
            cnt_W += blocks[i] == 'W';
            res = min(res, cnt_W);
        }
        return res;
    }
};

File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/Solution.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        res = cnt_w = blocks[:k].count("W")
        for a, b in zip(blocks, blocks[k:]):
            cnt_w += (b == "W") - (a == "W")
            res = min(res, cnt_w)
        return res
